src/classes/list-object.ts

```diff
@@ -0,0 +1,34 @@
+import { NS_RDF } from '../constants'
+
+export class ListObject {
+  s: string
+  p: string
+  graph: any
+  termFactory: any
+  NS_Rdf: string
+
+  constructor(s: string, p: string, graph: any, termFactory: any) {
+    this.s = s
+    this.p = p
+    this.graph = graph
+    this.termFactory = termFactory
+    this.NS_Rdf = NS_RDF
+  }
+  /**
+   *
+   * @param elt
+   * @param label
+   */
+  add(elt: any, label: string | undefined = undefined) {
+    let partLi = this.termFactory.blankNode(label)
+    this.graph.addQuad(this.s, this.p, partLi)
```

Comment:
Again we should make sure to comment each line of RDF work so developers can follow along. We don't need to comment repeat lines that are identical with different data, but we should explain what's going on.